src/mysql_conn.erl

```diff
@@ -408,7 +487,8 @@ handle_info(query_cache, #state{query_cache = Cache,
     {noreply, State#state{query_cache = Cache1}};
 handle_info({'DOWN', _MRef, _, Pid, _Info}, State) ->
     stop_server({application_process_died, Pid}, State);
-handle_info(ping, #state{socket = Socket, sockmod = SockMod} = State) ->
+handle_info(ping, #state{socket = Socket, sockmod = SockMod} = State)
+  when Socket =/= undefined ->
```

Comment:
Is this check needed? Ping is scheduled in post_connect, so we should always be connected here. If we're not connected, we have the problem that the ping is not re-scheduled...

Reply:
Hm, looks like a leftover from a previous iteration I did... You're right, it is not needed, I'll remove it.

Reply:
Removed in latest commit.